src/main/java/org/jenkinsci/plugins/scm_filter/GitHubPullRequestSkipTrait.java

```diff
@@ -0,0 +1,108 @@
+import hudson.plugins.git.GitSCM;
+import hudson.scm.SCMDescriptor;
+
+import jenkins.scm.api.SCMHead;
+import jenkins.scm.api.trait.*;
+
+import org.jenkinsci.plugins.github_branch_source.GitHubSCMBuilder;
+import org.jenkinsci.plugins.github_branch_source.GitHubSCMSourceRequest;
+import org.jenkinsci.plugins.github_branch_source.PullRequestSCMHead;
+
+import org.kohsuke.github.GHPullRequest;
+import org.kohsuke.github.GHRepository;
+import org.kohsuke.stapler.DataBoundConstructor;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.util.regex.Pattern;
+
+
+public class GitHubPullRequestSkipTrait extends SCMSourceTrait {
+
+
+    @DataBoundConstructor
+    public GitHubPullRequestSkipTrait () {
+    }
+
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected void decorateContext (SCMSourceContext<?, ?> context) {
+        context.withFilter(new GitHubPullRequestSkipTrait.ExcludePRsSCMHeadFilter());
+    }
+
+
+    /**
+     * Our descriptor.
+     */
+    @Extension
+    @SuppressWarnings("unused") // instantiated by Jenkins
+    public static class PullRequestSkipTraitDescriptorImpl extends SCMSourceTraitDescriptor {
+
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        @Nonnull
+        public String getDisplayName () {
+            return "Pull request title filtering behaviour";
+        }
+
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public boolean isApplicableToSCM (@Nonnull SCMDescriptor<?> scm) {
+            return scm instanceof GitSCM.DescriptorImpl;
+        }
+
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public boolean isApplicableToBuilder (@Nonnull Class<? extends SCMBuilder> builderClass) {
+            return GitHubSCMBuilder.class.isAssignableFrom(builderClass);
+        }
+    }
+
+
+    /**
+     * Filter that excludes pull requests according to its PR title
+     * if it contains [ci skip], [skip ci], or [wip], case insensitive.
+     * The words "ci" and "skip" can be separated by either a dash, an underscore, or a single space.
+     */
+    public static class ExcludePRsSCMHeadFilter extends SCMHeadFilter {
+
+
+        ExcludePRsSCMHeadFilter () {
+        }
+
+
+        @Override
+        public boolean isExcluded (
+                @Nonnull SCMSourceRequest scmSourceRequest,
+                @Nonnull SCMHead scmHead
+        ) throws IOException, InterruptedException {
+
+            if (scmHead instanceof PullRequestSCMHead) {
+                GHRepository repository = ((GitHubSCMSourceRequest) scmSourceRequest).getRepository();
+
+                // name for pull requests are formatted as "PR-<number>"
+                int prNumber = Integer.parseInt(scmHead.getName().substring(3));
+
+                GHPullRequest pull = repository.getPullRequest(prNumber);
+                return pull.getTitle().toLowerCase().matches(".*\\[(wip|ci[ -_]skip|skip[ -_]ci)\\].*");
```

Comment:
to ignore case, add `(?i)` at the beginning of the regular expression... 

Also.. I think we should expect `WIP`, `CI skip`, `skip ci` as the start of the title. 

Currently, with this regex, we allow it to be in the middle, because of the `.*` at the start of the regular expression